src/huggingface_hub/repository.py

```diff
@@ -404,6 +408,7 @@ def clone_from(self, repo_url: str, use_auth_token: Union[bool, str, None] = Non
         If this folder is a git repository with linked history, will try to update the repository.
         """
         token = use_auth_token if use_auth_token is not None else self.huggingface_token
+        repo_visibility = "public" if token is None else self.repo_visibility
```

Comment:
Maybe it's better to throw an error here so users don't accidentally publish a repo publicly.

Reply:
Added the error to the top